java/jdbc/src/main/java/io/vitess/jdbc/VitessResultSet.java

```diff
@@ -480,6 +480,13 @@ public Timestamp getTimestamp(int columnIndex) throws SQLException {
             return null;
         }
 
+        Constants.ZeroDateTimeBehavior zeroDateTimeBehavior =
+            vitessStatement.getConnection().getZeroDateTimeBehavior();
+        if (zeroDateTimeBehavior == Constants.ZeroDateTimeBehavior.CONVERTTONULL
+            && this.row.getRawValue(columnIndex).startsWith(Constants.ZERO_DATE_TIME_PREFIX)) {
```

Comment:
StartsWith .. and nothing more

Reply:
Yeah this mirrors the behavior of the mysql jdbc driver:
https://github.com/mysql/mysql-connector-j/blob/3289a357af6d09ecc1a10fd3c26e95183e5790ad/src/com/mysql/jdbc/ResultSetRow.java#L141